src/rust/aggregator/src/data/user_history.rs

```diff
@@ -49,27 +49,27 @@ impl Data for UserHistory {
             || self.last_indexed_timestamp.unwrap() + Duration::seconds(5) < Utc::now()
     }
 
-    async fn process_and_save_historical_data(&mut self) -> DataAggregationResult {
+    async fn process_and_save_historical_data(&mut self) -> PipelineAggregationResult {
         self.process_and_save_internal().await
     }
 
     fn poll_interval(&self) -> Option<std::time::Duration> {
         Some(std::time::Duration::from_secs(5))
     }
 
-    /// All database interactions are handled in a single atomic transaction. Processor insertions
+    /// All database interactions are handled in a single atomic transaction. Pipeline insertions
```

Comment:
I think this should have been left as the word processor

Because it is indeed referring to the submodule